pkg/pool-weighted/contracts/lbp/LBPool.sol

```diff
@@ -462,11 +247,31 @@ contract LBPool is ILBPool, WeightedPool, Ownable2Step, BaseHooks {
         // The WeightedPool will validate the starting weights (i.e., ensure they respect the minimum and sum to ONE).
         return
             NewPoolParams({
-                name: name,
-                symbol: symbol,
+                name: lbpCommonParams.name,
+                symbol: lbpCommonParams.symbol,
                 numTokens: _TWO_TOKENS,
                 normalizedWeights: normalizedWeights,
-                version: version
+                version: poolVersion
             });
     }
+
+    // Build and validate LBPCommonParams for initializing LBPCommon. Called on construction.
+    // Validates weights and adjusts startTime if needed.
+    function _buildLBPCommonParams(
+        LBPCommonParams memory lbpCommonParams,
+        LBPParams memory lbpParams
+    ) private view returns (LBPCommonParams memory finalCommonParams) {
+        finalCommonParams = lbpCommonParams;
+
+        // Checks that the weights are valid and `endTime` is after `startTime`. If `startTime` is in the past,
+        // avoid abrupt weight changes by overriding it with the current block time.
+        finalCommonParams.startTime = LBPoolLib.verifyWeightUpdateParameters(
```

Comment:
perhaps you can validate the start time at the common layer, and the weights inside `_buildWeightedPoolParams`?

Reply:
Rethinking this a bit, and all validations.

We're validating the migration at the factory, and mixing weights and start time here.

We should validate _everything_ at the factory level and either
- Leave comments where the variables are assigned
- Use a library and validate again to be extra careful
 
If there's a problem in the configuration and the pool creation reverts, it's incredibly annoying to debug because it will show as a create2 error everywhere, whereas if the factory reverts it's easy to understand what went wrong. If we validate a second time in the pool constructor we're extra safe, and that should never happen anyway if we do things right. WDYT?